hazelcast/src/hazelcast/client/spi/impl/ClientInvocation.cpp

```diff
@@ -29,96 +29,100 @@ namespace hazelcast {
         namespace spi {
             namespace impl {
                 ClientInvocation::ClientInvocation(spi::ClientContext &clientContext,
-                                                   std::auto_ptr<protocol::ClientMessage> &clientMessage,
+                                                   const boost::shared_ptr<protocol::ClientMessage> &clientMessage,
                                                    const std::string &objectName,
                                                    int partitionId) :
-                        ClientInvocationFuture(clientContext.getLogger()),
                         logger(clientContext.getLogger()),
                         lifecycleService(clientContext.getLifecycleService()),
                         clientClusterService(clientContext.getClientClusterService()),
                         invocationService(clientContext.getInvocationService()),
-                        executionService(clientContext.getClientExecutionService()),
+                        executionService(clientContext.getClientExecutionService().shared_from_this()),
                         clientMessage(boost::shared_ptr<protocol::ClientMessage>(clientMessage)),
```

Comment:
You already get a shared_ptr. This extra wrapping seems to be not necessary anymore. Or the change on the `ClientInvocation` constructor is not necessary.

Reply:
needs this since reference returned at here:
`impl::ClientExecutionServiceImpl &ClientContext::getClientExecutionService() const`

I will make another PR which will change the `ClientContext::getClientExecutionService()` method to return shared_ptr.